### DISCUSSION TITLE
Best practice to compile and debug iperf's code on Linux?

### DISCUSSION THREAD

**Discussion by alexisfrjp** [esnet/iperf#1721] [Q&A]
What's the common way to compile for debugging iperf?

I compiled it with `-g` but iperf3 is just a script and running directly the binary under `.src/.libs` always returns

> Usage: iperf3 [-s|-c host] [options]
> Try `iperf3 --help' for more information.

even though there are parameters.

Haven't found anything in `docs/dev.rst`

Thanks!

**Comment by swlars** [comment]
Hi, could you tell us a little more about how you're running iperf3 and what parameters you're using? 

Building a statically-linked executable might be your best option for using a debugger, but more information would be helpful.
